src/applications/vaos/tests/new-appointment/components/CommunityCareProviderSelectionPage/index.unit.spec.jsx

```diff
@@ -673,15 +669,26 @@ describe('VAOS <CommunityCareProviderSelectionPage>', () => {
     );
 
     // Choose Provider
-    userEvent.click(await screen.findByText(/Choose a provider/i));
-    userEvent.click(await screen.findByText(/use your current location/i));
+    userEvent.click(
+      await screen.findByRole('button', { name: /Choose a provider/i }),
+    );
+    // await waitFor(async () => {
```

Comment:
Just trying to get an understanding on  `await waitFor(async () => {})` function call.  When can we use/exclude the asynchronous call?
```suggestion
    await waitFor(async () => {
      expect(await screen.findByText(/Displaying 1 to/i)).to.be.ok;
    });
```

Reply:
Great question!!! I don't fully understand the difference between when to use 'await findBy*' vs 'waitFor' either (hence flaky tests). My understanding right now is they are the same but the use cases depend on the application. I found a couple of great articles that might shed some light on this.

My take right now is to use 'waitFor' when we need to wait for the page to settle to the final state. This typically happens when several async calls are made base on user reaction. Ex. click submit button. This is why I think you need to know the code to determine 'when' you should wait and 'what you should wait for. 

[https://dev.to/tipsy_dev/testing-library-writing-better-async-tests-c67](https://dev.to/tipsy_dev/testing-library-writing-better-async-tests-c67)
[https://kentcdodds.com/blog/common-mistakes-with-react-testing-library](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library)